dapp-oeth/src/hooks/useCurrencySwapper.js

```diff
@@ -140,7 +140,7 @@ const useCurrencySwapper = ({
       }
 
       const allowance = parseFloat(
-        allowances[coinNeedingApproval][allowanceCheckKey]
+        get(allowances, 'coinNeedingApproval[allowanceCheckKey]')
```

Comment:
This shouldn't work:
![Screenshot 2023-06-01 at 14 46 11](https://github.com/OriginProtocol/origin-dollar/assets/579910/96828071-db97-448d-a61e-8693cac75578)

Reply:
I think the correct code should be: 
```
get(allowances, ${coinNeedingApproval}.${allowanceCheckKey}')
```

in case both things in the array are object and you need to go 2 levels deep to get to the property.